functors_task_1/student/src/impl.cpp

```diff
@@ -1,2 +1,44 @@
 #include "impl.h"
 
+#include <algorithm>
+#include <forward_list>
+#include <map>
+#include <utility>
+
+BeerBrand BeerOrganizer::operator()()
+{
+    if (m_nextBeer <= static_cast<BaseType>(BeerBrand::None)
+            || (m_nextBeer >= static_cast<BaseType>(BeerBrand::Max)))
+    {
+        m_nextBeer = static_cast<BaseType>(BeerBrand::None) + 1;
+    }
+
+    return static_cast<BeerBrand>(m_nextBeer++);
+}
+
+bool isExpensiveBeer(const BeerBrand brand)
+{
+    //TODO: move it somewhere
+    static std::forward_list<BeerBrand> expensiveBeer = {BeerBrand::Corona, BeerBrand::HoeGaarden};
```

Comment:
Wow, please implement this task without containers usage. Especially avoid lists (why - we'll talk about it at the second part of the training).

Don't overcomplicate here.

Reply:
Yes, the forward list shouldn't be used here.

And the one more thing. How I should guess that it is forbidden to use containers? :)